holidays/countries/mongolia.py

```diff
@@ -0,0 +1,146 @@
+        self._add_womens_day(tr("Олон улсын эмэгтэйчүүдийн өдөр"))
+
+        # Children's Day.
+        self._add_childrens_day(tr("Хүүхдийн баяр"))
+
+        # Established on December 20th, 2019.
+        if self._year >= 2020:
+            # The Buddha's Birthday.
+            self._add_buddha_day(tr("Бурхан багшийн Их дүйчин өдөр"))
+
+        # Naadam.
+        self._add_holiday_jul_11(tr("Наадам"))
+
+        # Naadam Holiday.
+        name = tr("Наадмын баяр")
+        self._add_holiday_jul_12(name)
+        self._add_holiday_jul_13(name)
+        self._add_holiday_jul_14(name)
+        self._add_holiday_jul_15(name)
+
+        # Established on November 8th, 2012.
+        if self._year >= 2012:
+            # Genghis Khan's Birthday.
+            self._add_genghis_khan_day(tr("Их Эзэн Чингис хааны өдөр"))
+
+        # Established on November 18th, 2016.
+        if self._year >= 2016:
+            # Republic Day.
+            self._add_holiday_nov_26(tr("Бүгд Найрамдах Улс тунхагласан өдөр"))
+
+        # Established on August 16th, 2007.
+        # Renamed on December 23rd, 2011.
+        if self._year >= 2007:
+            self._add_holiday_dec_29(
+                # Restoration of Freedom and Independence Day.
+                tr("Үндэсний эрх чөлөө, тусгаар тогтнолоо сэргээсний баярын өдөр")
+                if self._year >= 2011
+                # Independence Day.
+                else tr("Үндэсний эрх чөлөөний өдөр")
+            )
+
+    def _populate_workday_holidays(self):
+        # Constitutional Day of Mongolia.
+        self._add_holiday_jan_13(tr("Монгол Улсын Үндсэн хуулийн өдөр"))
+
+        # Patriots' Day.
+        self._add_holiday_mar_1(tr("Эх орончдын өдөр"))
+
+        if self._year >= 2011:
+            # Mongolia Military Day.
+            self._add_holiday_mar_18(tr("Монгол цэргийн өдөр"))
+
+        # Health Protection Day.
+        self._add_holiday_apr_7(tr("Эрүүл мэндийг хамгаалах өдөр"))
+
+        # Intellectual Property Day.
+        self._add_holiday_apr_26(tr("Оюуны өмчийг хамгаалах өдөр"))
+
+        # Family Day.
+        self._add_holiday_may_15(tr("Гэр бүлийн өдөр"))
+
+        if self._year >= 2009:
+            # Mongolia National Flag Day.
+            self._add_holiday_jul_10(tr("Монгол Улсын төрийн далбааны өдөр"))
+
+        # Youth Day.
+        self._add_holiday_aug_25(tr("Залуучуудын өдөр"))
+
+        # Memorial Day of Political Defendants.
+        self._add_holiday_sep_10(tr("Улс төрийн хэлмэгдэгсдийн дурсгалын өдөр"))
+
+        # Elders' Day.
+        self._add_holiday_oct_1(tr("Ахмадын өдөр"))
+
+        # Mongolia Capital Day.
+        self._add_holiday_oct_29(tr("Монгол Улсын Нийслэлийн өдөр"))
+
+        if 2012 <= self._year <= 2015:
+            # Republic Day.
+            self._add_holiday_nov_26(tr("Бүгд Найрамдах Улс тунхагласан өдөр"))
+
+        # Democracy and Human Rights Day.
+        self._add_holiday_dec_10(tr("Ардчилал, хүний эрхийн өдөр"))
+
```

Comment:
https://legalinfo.mn/mn/detail?lawId=100713&showType=1
https://legalinfo.mn/mn/detail?lawId=100710&showType=1
```suggestion
        # Established on August 16th, 2007.
        # Renamed and considered a Public Holiday on December 23rd, 2011.
        if 2007<= self._year <= 2010:
            # Independence Day.
            self._add_holiday_dec_29(tr("Үндэсний эрх чөлөөний өдөр"))

```